aptos-move/framework/aptos-framework/sources/primary_fungible_store.move

```diff
@@ -126,6 +126,8 @@ module aptos_framework::primary_fungible_store {
     /// Withdraw `amount` of fungible asset from the given account's primary store.
     public fun withdraw<T: key>(owner: &signer, metadata: Object<T>, amount: u64): FungibleAsset {
         let store = primary_store(signer::address_of(owner), metadata);
+        // Check if the store object has been burnt or not. If so, unburn it first.
+        may_be_unburn(owner, store);
```

Comment:
Why is this necessary? it seems fine to error out in this case.

Reply:
To make it seem more seamless to transition between burn and unburn